eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -45,14 +47,26 @@ foreach ($branch in $branches)
   $openPullRequests = $pullRequests | ? { $_.State -eq "open" }
   if ($openPullRequests.Count -gt 0)
   {
-    LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has open pull Requests. Skipping"
+    LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has open pull Requests. Skipping"
     LogDebug $openPullRequests.url
     continue
   }
 
-  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
+  if ($LastCommitOlderThan) {
+    if (!$res.object -or !$res.object.url) {
+      LogWarning "No commit url returned from response. Skipping... "
+      continue
+    }
+    $commitDate = Get-GithubBranchLastCommitDate -commitUrl $res.object.url
+    if ($commitDate -gt $LastCommitOlderThan) {
```

Comment:
Are you sure this is doing a Date comparison or is it doing a string comparison? We should probably cast these to datetime objects to ensure it is a date comparison and you will want to also check to ensure $commitDate isn't null.

Reply:
$commitDate and $LastCommitOlderThan are both datetime type.
<img width="460" alt="image" src="https://user-images.githubusercontent.com/48036328/165403627-aafbcbd1-3098-4768-b73a-bb5c6444fe12.png">
<img width="453" alt="image" src="https://user-images.githubusercontent.com/48036328/165403661-598d4352-9e21-4701-a77e-e0a47a062c87.png">

Powershell can compare the DateTime type with `-gt`, `-lt`. [Ref](https://docs.microsoft.com/en-us/answers/questions/189447/comparing-dates-with-powershell.html)